internal/databases/postgres/pagefile.go

```diff
@@ -107,7 +107,6 @@ func isPagedFile(info os.FileInfo, filePath string) bool {
 	if info.IsDir() ||
 		((!strings.Contains(filePath, DefaultTablespace)) && (!strings.Contains(filePath, NonDefaultTablespace))) ||
 		info.Size() == 0 ||
-		info.Size()%DatabasePageSize != 0 ||
```

Comment:
Hm, `isPagedFile` is actually being used not only in the paged file verification process, but in the delta backups logic too. You introduced the check for unexpected EOF for paged file verifier, but is it safe in other places too? I am not sure if the delta file creation logic is proof to the non-multiple of page size files.

Reply:
yes, i do some simple real live test and you are absolutely right in your doubts.
1. create DB -> create table -> fill data on it
2. do FULL BACKUP with verify checksum
3. currupted data in table 
4. do INCREMENT backup with verify checksum
5. restore backup

In fact, a file with "other data" has been restored, but it is necessary to restore one to one (bytes to bytes).

------------------RESULTS -------------------------
**FILE /data/pg_data/postgres/base/16493/16628 before RESTORE - 8195 bytes**
[root@mbogatyrev-test32-db-01 16493]# hexdump -C /data/pg_data/postgres/base/16493/16628
00000000  00 00 00 00 b8 a1 02 e4  f3 2f 00 00 24 00 88 1f  |........./..$...|
00000010  00 20 04 20 00 00 00 00  d8 9f 44 00 b0 9f 42 00  |. . ......D...B.|
00000020  88 9f 44 00 00 00 00 00  00 00 00 00 00 00 00 00  |..D.............|
00000030  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
*
00001f80  00 00 00 00 00 00 00 00  33 03 00 00 00 00 00 00  |........3.......|
00001f90  07 00 00 00 00 00 00 00  03 00 02 00 02 08 18 00  |................|
00001fa0  03 00 00 00 0d 4d 49 53  48 41 00 00 00 00 00 00  |.....MISHA......|
00001fb0  33 03 00 00 00 00 00 00  06 00 00 00 00 00 00 00  |3...............|
00001fc0  02 00 02 00 02 08 18 00  02 00 00 00 0b 4f 4c 47  |.............OLG|
00001fd0  41 00 00 00 00 00 00 00  33 03 00 00 00 00 00 00  |A.......3.......|
00001fe0  05 00 00 00 00 00 00 00  01 00 02 00 02 08 18 00  |................|
00001ff0  01 00 00 00 0d 4d 49 4b  45 31 00 00 00 48 65 6c  |.....MIKE1...Hel|
00002000  6c 6f 21                                          |lo!|
00002003

**FILE /data/pg_data/postgres/base/16493/16628 AFTER RESTORE - 8195 bytes too but with empty data at the end**
[root@mbogatyrev-test32-db-01 16493]# hexdump -C /data/pg_backup/full_fuck/base/16493/16628
00000000  00 00 00 00 b8 a1 02 e4  f3 2f 00 00 24 00 88 1f  |........./..$...|
00000010  00 20 04 20 00 00 00 00  d8 9f 44 00 b0 9f 42 00  |. . ......D...B.|
00000020  88 9f 44 00 00 00 00 00  00 00 00 00 00 00 00 00  |..D.............|
00000030  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
*
00001f80  00 00 00 00 00 00 00 00  33 03 00 00 00 00 00 00  |........3.......|
00001f90  07 00 00 00 00 00 00 00  03 00 02 00 02 08 18 00  |................|
00001fa0  03 00 00 00 0d 4d 49 53  48 41 00 00 00 00 00 00  |.....MISHA......|
00001fb0  33 03 00 00 00 00 00 00  06 00 00 00 00 00 00 00  |3...............|
00001fc0  02 00 02 00 02 08 18 00  02 00 00 00 0b 4f 4c 47  |.............OLG|
00001fd0  41 00 00 00 00 00 00 00  33 03 00 00 00 00 00 00  |A.......3.......|
00001fe0  05 00 00 00 00 00 00 00  01 00 02 00 02 08 18 00  |................|
00001ff0  01 00 00 00 0d 4d 49 4b  45 31 00 00 00 00 00 00  |.....MIKE1......|
00002000  00 00 00                                          |...|
00002003

Reply:
Sorry for the delay.
@usernamedt so do we want short-read files saved as paged or as non-paged in delta backups?

Reply:
> Sorry for the delay. @usernamedt so do we want short-read files saved as paged or as non-paged in delta backups?

Sorry for the delay [2] :) 

I think that we should retain the current behavior for the delta backups and alter only the verification part.

@boosterKRD one way to solve this is the two functions:
1. IsPagedFile() - for checksum verficiation
2. IsAlignedPagedFile() - for delta backups